lib/OpenQA/Task/Asset/Limit.pm

```diff
@@ -45,6 +46,7 @@ sub _update_exclusively_kept_asset_size {
 
 sub _limit {
     my ($app, $job) = @_;
+    my $ensure_task_retry_on_termination_signal_guard = OpenQA::Task::SignalGuard->new($job);
```

Comment:
This looks very odd to me, considering the variable will never be used and looks as though it should go out of scope. Except `weaken` and `$SIG` change that.
You're not introducing this API here, so I'm just mentioning it for the record.

Reply:
Having an unused variable is how it works with guards in Perl (and C++ or any language utilizing destructors).